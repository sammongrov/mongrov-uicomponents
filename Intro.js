import React, { Component } from 'react';
import { View } from 'react-native';
import { Icon } from '@ui/components';
import AppIntroSlider from 'react-native-app-intro-slider';
import PropTypes from 'prop-types';
import { styles } from 'react-native-theme';

export default class Intro extends Component {
  _renderNextButton = () => (
    <View style={styles.btnIconRounded}>
      <Icon
        testID="next-button"
        name="arrow-right"
        type="material-community"
        color="rgba(255, 255, 255, .9)"
        size={24}
      />
    </View>
  );

  _renderDoneButton = () => (
    <View style={styles.btnIconRounded}>
      <Icon
        testID="done-button"
        name="check"
        type="material-community"
        color="rgba(255, 255, 255, .9)"
        size={24}
      />
    </View>
  );

  render() {
    const { slides, onDone, onSkip } = this.props;
    return (
      <AppIntroSlider
        testID="guides"
        slides={slides}
        onDone={onDone}
        onSkip={onSkip}
        renderNextButton={this._renderNextButton}
        renderDoneButton={this._renderDoneButton}
        showSkipButton
      />
    );
  }
}
Intro.propTypes = {
  slides: PropTypes.arrayOf(PropTypes.object).isRequired,
  onDone: PropTypes.func.isRequired,
  onSkip: PropTypes.func.isRequired,
};
