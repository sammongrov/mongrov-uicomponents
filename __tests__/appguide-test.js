import { Alert } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Intro from '../Intro';

const onDone = () => Alert.alert('Done');
const onSkip = () => Alert.alert('Skip');

it('appguide renders correctly', () => {
  const intro = [
    {
      key: 'slide1',
      title: 'Cost = Asset',
      text: 'Like banking, dApps hold some tokens for a free scalable cloud',
      image: {
        uri: 'https://d30y9cdsu7xlg0.cloudfront.net/png/1266818-200.png',
      },
      backgroundColor: '#59b2ab',
    },
    {
      key: 'slide2',
      title: 'Fair price mining',
      text: 'Fair market price for Miners, Sharders, Blobbers',
      image: {
        uri: 'https://d30y9cdsu7xlg0.cloudfront.net/png/215031-200.png',
      },
      backgroundColor: '#febe29',
    },
    {
      key: 'slide3',
      title: 'Efficient infrastructure',
      text: 'Decoupled CPU, RAM, and SSD needs for cost-efficient mining, sharding and blobbing',
      image: {
        uri: 'https://d30y9cdsu7xlg0.cloudfront.net/png/1484653-200.png',
      },
      backgroundColor: '#4F00BC',
    },
  ];
  const tree = renderer.create(<Intro slides={intro} onDone={onDone} onSkip={onSkip} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('icon button renders correctly', () => {
  const intro = [
    {
      key: 'slide1',
      title: 'Cost = Asset',
      text: 'Like banking, dApps hold some tokens for a free scalable cloud',
      image: {
        uri: 'https://d30y9cdsu7xlg0.cloudfront.net/png/1266818-200.png',
      },
      backgroundColor: '#59b2ab',
    },
    {
      key: 'slide2',
      title: 'Fair price mining',
      text: 'Fair market price for Miners, Sharders, Blobbers',
      image: {
        uri: 'https://d30y9cdsu7xlg0.cloudfront.net/png/215031-200.png',
      },
      backgroundColor: '#febe29',
    },
    {
      key: 'slide3',
      title: 'Efficient infrastructure',
      text: 'Decoupled CPU, RAM, and SSD needs for cost-efficient mining, sharding and blobbing',
      image: {
        uri: 'https://d30y9cdsu7xlg0.cloudfront.net/png/1484653-200.png',
      },
      backgroundColor: '#4F00BC',
    },
  ];
  const tree = renderer.create(<Intro slides={intro} onDone={onDone} onSkip={onSkip} />);
  const inst = tree.getInstance();
  expect(inst._renderDoneButton()).toMatchSnapshot();
});
