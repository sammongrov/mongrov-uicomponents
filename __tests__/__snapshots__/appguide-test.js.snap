// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appguide renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bounces={false}
    data={
      Array [
        Object {
          "backgroundColor": "#59b2ab",
          "image": Object {
            "uri": "https://d30y9cdsu7xlg0.cloudfront.net/png/1266818-200.png",
          },
          "key": "slide1",
          "text": "Like banking, dApps hold some tokens for a free scalable cloud",
          "title": "Cost = Asset",
        },
        Object {
          "backgroundColor": "#febe29",
          "image": Object {
            "uri": "https://d30y9cdsu7xlg0.cloudfront.net/png/215031-200.png",
          },
          "key": "slide2",
          "text": "Fair market price for Miners, Sharders, Blobbers",
          "title": "Fair price mining",
        },
        Object {
          "backgroundColor": "#4F00BC",
          "image": Object {
            "uri": "https://d30y9cdsu7xlg0.cloudfront.net/png/1484653-200.png",
          },
          "key": "slide3",
          "text": "Decoupled CPU, RAM, and SSD needs for cost-efficient mining, sharding and blobbing",
          "title": "Efficient infrastructure",
        },
      ]
    }
    disableVirtualization={false}
    extraData={750}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onDone={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onSkip={[Function]}
    pagingEnabled={true}
    renderDoneButton={[Function]}
    renderItem={[Function]}
    renderNextButton={[Function]}
    scrollEventThrottle={50}
    showSkipButton={true}
    showsHorizontalScrollIndicator={false}
    slides={
      Array [
        Object {
          "backgroundColor": "#59b2ab",
          "image": Object {
            "uri": "https://d30y9cdsu7xlg0.cloudfront.net/png/1266818-200.png",
          },
          "key": "slide1",
          "text": "Like banking, dApps hold some tokens for a free scalable cloud",
          "title": "Cost = Asset",
        },
        Object {
          "backgroundColor": "#febe29",
          "image": Object {
            "uri": "https://d30y9cdsu7xlg0.cloudfront.net/png/215031-200.png",
          },
          "key": "slide2",
          "text": "Fair market price for Miners, Sharders, Blobbers",
          "title": "Fair price mining",
        },
        Object {
          "backgroundColor": "#4F00BC",
          "image": Object {
            "uri": "https://d30y9cdsu7xlg0.cloudfront.net/png/1484653-200.png",
          },
          "key": "slide3",
          "text": "Decoupled CPU, RAM, and SSD needs for cost-efficient mining, sharding and blobbing",
          "title": "Efficient infrastructure",
        },
      ]
    }
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="guides"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "space-around",
              },
              Object {
                "backgroundColor": "#59b2ab",
                "height": 1334,
                "paddingBottom": 64,
                "paddingTop": 20,
                "width": 750,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, .7)",
                  "fontSize": 26,
                  "fontWeight": "300",
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            Cost = Asset
          </Text>
          <Image
            source={
              Object {
                "uri": "https://d30y9cdsu7xlg0.cloudfront.net/png/1266818-200.png",
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, .7)",
                  "fontSize": 16,
                  "fontWeight": "300",
                  "paddingHorizontal": 16,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Like banking, dApps hold some tokens for a free scalable cloud
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "space-around",
              },
              Object {
                "backgroundColor": "#febe29",
                "height": 1334,
                "paddingBottom": 64,
                "paddingTop": 20,
                "width": 750,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, .7)",
                  "fontSize": 26,
                  "fontWeight": "300",
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            Fair price mining
          </Text>
          <Image
            source={
              Object {
                "uri": "https://d30y9cdsu7xlg0.cloudfront.net/png/215031-200.png",
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, .7)",
                  "fontSize": 16,
                  "fontWeight": "300",
                  "paddingHorizontal": 16,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Fair market price for Miners, Sharders, Blobbers
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "space-around",
              },
              Object {
                "backgroundColor": "#4F00BC",
                "height": 1334,
                "paddingBottom": 64,
                "paddingTop": 20,
                "width": 750,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, .7)",
                  "fontSize": 26,
                  "fontWeight": "300",
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            Efficient infrastructure
          </Text>
          <Image
            source={
              Object {
                "uri": "https://d30y9cdsu7xlg0.cloudfront.net/png/1484653-200.png",
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, .7)",
                  "fontSize": 16,
                  "fontWeight": "300",
                  "paddingHorizontal": 16,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Decoupled CPU, RAM, and SSD needs for cost-efficient mining, sharding and blobbing
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 16,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 16,
          "justifyContent": "center",
          "margin": 16,
        }
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "position": "absolute",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 18,
                  "padding": 12,
                },
                null,
              ]
            }
          >
            Skip
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "width": 10,
            },
            Object {
              "backgroundColor": "rgba(255, 255, 255, .9)",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "width": 10,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, .2)",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "width": 10,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, .2)",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .9)",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="next-button"
              type="material-community"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`icon button renders correctly 1`] = `
<Component>
  <Icon
    allowFontScaling={false}
    color="rgba(255, 255, 255, .9)"
    name="check"
    size={24}
    testID="done-button"
    type="material-community"
  />
</Component>
`;
